### DISCUSSION TITLE
how to update forge?

### DISCUSSION THREAD

**Discussion by fordragon88** [lllyasviel/stable-diffusion-webui-forge#329] [Q&A] [answered]
how to update forge, I installed it last week, but I see you already have a newer version

**Comment by VeteranXT** [comment]
where is your repository is do following CMD > Git clone in same folder.  Or if run update.bat if you downloaded on click installer.

**Reply by fordragon88** [reply]
I already have it installed via cmd, how can I update it, I used to write to A1111 before git pull, and it was updated

**Reply by VeteranXT** [reply]
Then you already have?

**Comment by freecoderwaifu** [comment]
Open CMD inside the Forge folder then type `git pull` 

It's an universal command to update any git repo (what most Stable Diffusion UIs are hosted on, including Forge).

If you see `Already up to date.` at the top of the CMD window then it's up to date.

Add it after `@echo off` in your `webui-user.bat` if you want it to auto update on every launch, just remember you manually added it to avoid "wtf why did it update on its own?????" moments. 

Add `set COMMANDLINE_ARGS= --update-all-extensions` if you also want to auto update extensions on launch, note this only applies to third party extensions and not built in ones (the ones that say built-in in the Extensions tab), as those are updated with git pull.

**Comment by giovanith** [comment]
git pull... returns to me : 

fatal: not a git repository (or any of the parent directories): .git

how to solve it ? I cannot update forge

**Reply by philmcrakin** [reply]
That message is typically given when you aren't in the parent directory of a git repository.
Make sure you are in the root directory of your forge installation. 

For you that may be: **C:\stable-diffusion-webui** or **C:\stable-diffusion-webui-forge** depending on your OS and if you customized install path.

For me its at: **/home/username/stable-diffusion-webui/**

There should be a .git and .github directory and many others inside your git repositories parent directory.

Make sure you are running the _**git pull**_ command from the parent directory and not one of the nested directories.

**Reply by fordragon88** [reply, marked as answer]
I figured it out

step 1. open the directory /stable-diffusion-webui-forge
step 2. copy the file webui-user.bat , somewhere in another directory
step 3. Open CMD inside the Forge folder(/stable-diffusion-webui-forge) then type 
git stash
git pull
git stash pop
step 4. copy back the file webui-user.bat to folder(/stable-diffusion-webui-forge)

**Reply by DrJimmyBrungus** [reply]
I'm not too familiar with Git - could anyone explain how is the above different from doing just git pull by itself?

How would you format the above into a .bat so I can update in one click?

**Reply by VeteranXT** [reply]
Git clone = Clone specific repo with latest commit
Git pull = Get latest commit (get update)
Git Checkout (specific commit) = Get specific version
Git Stash = Stash your changes for later
Git Stash pop = Stash your changes for later
